x-pack/libbeat/reader/parquet/parquet.go

```diff
@@ -9,85 +9,110 @@ import (
 	"context"
 	"fmt"
 	"io"
+	"time"
 
 	"github.com/apache/arrow/go/v14/arrow/memory"
 	"github.com/apache/arrow/go/v14/parquet"
 	"github.com/apache/arrow/go/v14/parquet/file"
 	"github.com/apache/arrow/go/v14/parquet/pqarrow"
+	"github.com/elastic/elastic-agent-libs/logp"
 )
 
 // BufferedReader parses parquet inputs from io streams.
 type BufferedReader struct {
 	cfg          *Config
 	fileReader   *file.Reader
 	recordReader pqarrow.RecordReader
+	log          *logp.Logger
 }
 
 // NewBufferedReader creates a new reader that can decode parquet data from an io.Reader.
 // It will return an error if the parquet data stream cannot be read.
 // Note: As io.ReadAll is used, the entire data stream would be read into memory, so very large data streams
 // may cause memory bottleneck issues.
 func NewBufferedReader(r io.Reader, cfg *Config) (*BufferedReader, error) {
-	batchSize := 1
-	if cfg.BatchSize > 1 {
-		batchSize = cfg.BatchSize
+	log := initLogger()
+	start := time.Now()
```

Comment:
Do we need this since the logging will have a timestamp?

Reply:
Yea true, but sometimes calculating the time via timestamps can get a bit cumbersome, hence a clear duration value could help make things simpler to parse

Reply:
The consequence is fruitless work for all cases where DEBUG has not been selected. I'm not sure it's worth it.

Reply:
Sometimes judging the time via the timestamp can get cumbersome, hence wanted a clear duration marker.

Reply:
What would the duration tell you that you would need to know when debugging?

Reply:
Sorry, the same message got sent twice due to a network issue on my end. So this init would happen only once, I've removed the duration from the reader. Only purpose this would serve is QoL improvements fro the person debugging. It easier to parse than looking at logging timestamps for some.

Reply:
@efd6, I've addressed the remaining suggestions, I'll remove the duration here if you still feel my latest QoL point does not justify this.

Reply:
I'm not convinced that it's helpful, but if you want to keep it, you can after changing how the values are rendered. At the moment, the calls will always allocate a string and perform the relatively expensive `fmt.Sprintf` call. From the zap code, there should not be a need to call `fmt.Sprintf`; e.g. `log.Debugw("initialization process completed", "duration", time.Since(start))` should work.

Reply:
Ok I'll remove the duration as a whole for now and if down the line people feel it would be helpful to have it, I'll add it back based on feedback.

Reply:
@efd6, removed the duration tracking